apps/designer/app/designer/shared/design-tokens-manager/schema.ts

```diff
@@ -1,34 +1,29 @@
 import { z } from "zod";
 import { groups } from "./groups";
 
-type UnionToIntersection<U> = (
-  U extends never ? never : (arg: U) => never
-) extends (arg: infer I) => void
-  ? I
-  : never;
+const mapToProp = <
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends ReadonlyArray<any>,
+  Prop extends keyof T[number]
+>(
+  arr: readonly [...T],
+  prop: Prop
+): {
+  [K in keyof T]: T[K][Prop];
+} => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return arr.map((item) => item[prop]) as any;
+};
 
-type UnionToTuple<T> = UnionToIntersection<
-  T extends never ? never : (t: T) => T
-> extends (_: never) => infer W
-  ? [...UnionToTuple<Exclude<T, W>>, W]
-  : [];
-
-type Type = typeof groups[number]["type"];
-type Group = typeof groups[number]["group"];
-
-// zod is super restrictive on enum type https://github.com/colinhacks/zod/issues/1118
-// so we convert union to tuple here its seriously safer than any as it will break on things like
-// ['a', 'b'] as UnionToTuple<Type> etc
-const types = groups.map((group) => group.type) as UnionToTuple<Type>;
-const groupNames = groups.map((group) => group.group) as UnionToTuple<Group>;
```

Comment:
This was fine here as those variables don't escape this module and are only used as inputs for zod. However, `UnionToTuple` is not safe in the generic sense - the order of the elements in the result is **unstable** and can't be relied upon.

Therefore, I've replaced this with a different approach - mostly just for educational reasons, to showcase how this kind of a problem can be approached in the future etc.